core/unit_test/TestGraph.hpp

```diff
@@ -864,6 +900,118 @@ TEST(TEST_CATEGORY, when_all_type) {
   static_assert(std::is_same_v<decltype(tail), node_ref_tail_t>);
 }
 
+#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP)
+template <GraphNodeType value, typename T>
+__global__ void set_to(T* const data) {
+  data[threadIdx.y] += static_cast<T>(value);
+}
+#endif
+
+template <typename Exec, typename ViewType>
+struct ExternalCapture {
```

Comment:
Another name could be `BackendCapture`. It's fine to keep it as is too.

Reply:
I chose `External` because capture would be used to integrate "external" libraries into a graph.